app/lib/webfinger_resource.rb

```diff
@@ -0,0 +1,60 @@
+class WebfingerResource
+  attr_reader :resource
+
+  def initialize(resource)
+    @resource = resource
+  end
+
+  def username
+    case resource
+    when /\Ahttps?/i
+      username_from_url
+    when /acct/
```

Comment:
I should have submitted that as a change request - this should be able to handle `username@domain` without the `acct:` at the front